### ISSUE TITLE
Inconsistent Focus Management in Admin Interface

### ISSUE THREAD

**Issue by avaaec** [pressbooks/pressbooks#1770]
Edited to include extra context/details:

**Pages Affected:**
-  Add Part
-  Add Chapter
-  Add Front Matter
-  Add Back Matter

**Steps to Reproduce**
1.  Go to Admin interface for a book
2.  Navigate to "My Catalog" link in the Dashboard submenu in the Main Menu navigation region
3.  Notice on activation the first tab stop brings a user to “Skip to main content”
4.  Navigate to "Add Part" link under "Organize" sub menu in the Main Menu navigation region
5.  Notice that on activation the users focus is put on the "Add Title" Field of the Main region

**Expected behavior:**
According to WCAG guideline 2.4.3 Focus Order, “If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability.” 

It is expected that all of the controls in the main menu for the authoring region behave in predictable and consistent ways. More specifically, on pages affected, focus is brought to the first field on the page instead of the start of the page. Whether focus is brought back to the start of the page, left on the control, or moved to the Main region, the behavior should be the same for all controls.

**Actual behavior:**
Different links in the Admin Main Menu navigation region manage focus in different and inconsistent ways. This causes confusion for keyboard and non-visual users. When activating a control in the Main Menu that the user has not activated before they will expect their focus to be managed in the same way as with previous controls, if/when this expectation is broken they will likely be confused. 

Further, WCAG's reasons for maintaining consistent focus order across pages:

> People with mobility impairments who must rely on keyboard access for operating a page benefit from a logical, usable focus order. 
> People with disabilities that make reading difficult can become disoriented when tabbing takes focus someplace unexpected. They benefit from a logical focus order.
> People with visual impairments can become disoriented when tabbing takes focus someplace unexpected or when they cannot easily find the content surrounding an interactive element.
> Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.
**Source:**  [“Focus Order:Understanding SC 2.4.3.”](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

**Comment by avaaec** [comment]
This is UW priority 2 issue

**Comment by SteelWagstaff** [comment]
Pages Affected:
-     Add Part
-     Add Chapter
-     Add Front Matter
-     Add Back Matter

**Steps to Reproduce**
1.  Go to Admin interface for a book
2.  Navigate to "My Catalog" link in the Dashboard submenu in the Main Menu navigation region
3.  Notice on activation the first tab stop brings a user to “Skip to main content”
4.  Navigate to "Add Part" link under "Organize" sub menu in the Main Menu navigation region
5.  Notice that on activation the users focus is put on the "Add Title" Field of the Main region

**Expected behavior:**
According to WCAG guideline 2.4.3 Focus Order, “If a Web page can be navigated sequentially and the navigation sequences affect meaning or operation, focusable components receive focus in an order that preserves meaning and operability.” 

It is expected that all of the controls in the main menu for the authoring region behave in predictable and consistent ways. More specifically, on pages affected, focus is brought to the first field on the page instead of the start of the page. Whether focus is brought back to the start of the page, left on the control, or moved to the Main region, the behavior should be the same for all controls.

**Actual behavior:**
Different links in the Admin Main Menu navigation region manage focus in different and inconsistent ways. This causes confusion for keyboard and non-visual users. When activating a control in the Main Menu that the user has not activated before they will expect their focus to be managed in the same way as with previous controls, if/when this expectation is broken they will likely be confused. 

Further, WCAG's reasons for maintaining consistent focus order across pages:

> People with mobility impairments who must rely on keyboard access for operating a page benefit from a logical, usable focus order. 
> People with disabilities that make reading difficult can become disoriented when tabbing takes focus someplace unexpected. They benefit from a logical focus order.
> People with visual impairments can become disoriented when tabbing takes focus someplace unexpected or when they cannot easily find the content surrounding an interactive element.
> Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.
**Source:**  [“Focus Order:Understanding SC 2.4.3.”](www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html) Understanding Success Criterion 2.4.3 | Understanding WCAG 2.0,

**Comment by SteelWagstaff** [comment]
We've taken a closer look at this issue and discussed with an outside accessibility consultant. We're going to close without further changes because focus behaviour in Pressbooks is applied in a consistent and logical way. To be more specific, in all parts of the interface where the first step is the creation of new content (for all pages where the page title begins with 'Add [Part/Chapter/etc]'), the focus is set to the first editable field. This focus decision is consistent throughout Pressbooks, and we believe supports the user's probable intention when they choose to add new content. The My Catalog page manages existing content, so the behaviour is different (similar to Organize, Export, etc).
